website_docs/getting-started.md

```diff
@@ -3,552 +3,897 @@ title: "Getting Started"
+// Wrap the handleRollDice so that telemetry data
+// can be automatically generated for it
+func wrapHandler() {
+	handler := http.HandlerFunc(handleRollDice)
+	wrappedHandler := otelhttp.NewHandler(handler, "rolldice")
+	http.Handle("/rolldice", wrappedHandler)
+}
 
-To back up a bit, a trace is a type of telemetry that represents work being done by a service. A trace is a record of the connection(s) between participants processing a transaction, often through client/server requests processing and other forms of communication.
+func main() {
+	ctx := context.Background()
 
-Each part of the work that a service performs is represented in the trace by a span. Those spans are not just an unordered collection. Like the call stack of our application, those spans are defined with relationships to one another. The "root" span is the only span without a parent, it represents how a service request is started. All other spans have a parent relationship to another span in the same trace.
+	tp := newTracerProvider()
+	defer func() { _ = tp.Shutdown(ctx) }()
 
-If this last part about span relationships doesn't make complete sense now, don't worry. The most important takeaway is that each part of your code, which does some work, should be represented as a span. You will have a better understanding of these span relationships after you instrument your code, so let's get started.
+	otel.SetTracerProvider(tp)
 
-Start by instrumenting the `Run` method.
+	// Register context and baggage propagation.
+	// Although not strictly necessary, for this sample,
+	// it is required for distributed tracing.
+	otel.SetTextMapPropagator(
+		propagation.NewCompositeTextMapPropagator(
+			propagation.TraceContext{},
+			propagation.Baggage{},
+		),
+	)
 
-```go
-// Run starts polling users for Fibonacci number requests and writes results.
-func (a *App) Run(ctx context.Context) error {
-	for {
-		// Each execution of the run loop, we should get a new "root" span and context.
-		newCtx, span := otel.Tracer(name).Start(ctx, "Run")
-
-		n, err := a.Poll(newCtx)
-		if err != nil {
-			span.End()
-			return err
-		}
-
-		a.Write(newCtx, n)
-		span.End()
+	wrapHandler()
+
+	err := http.ListenAndServe(":8080", nil)
+	if err != nil {
+		log.Fatal(err)
 	}
 }
 ```
 
-The above code creates a span for every iteration of the for loop. The span is created using a [`Tracer`] from the [global `TracerProvider`](https://pkg.go.dev/go.opentelemetry.io/otel#GetTracerProvider). You will learn more about [`TracerProvider`]s and handle the other side of setting up a global [`TracerProvider`] when you install an SDK in a later section. For now, as an instrumentation author, all you need to worry about is that you are using an appropriately named [`Tracer`] from a [`TracerProvider`] when you write `otel.Tracer(name)`.
-
-Next, instrument the `Poll` method.
+This will initialize tracing and wrap the HTTP handler so that tracing data can
```

Comment:
Highlight the key pieces of code which make this possible

Reply:
Heh, this one might be a bit of a doozy. I put in a bit of an explainer after the code sample. Part of the reason why I didn't include this just yet is because it's...a lot of dense, difficult code. Otel-go init code is pretty gnarly. But I also think it's good to set someone up well with the "right" stuff that's copy-pastable.

What do you think of the updated changes?

Reply:
Its an improvement but you could describe why Resource, TracerProvider etc. are required.

Reply:
Okay, I added inline code comments about resources specifically, then also summarized the _why_ of both `Resource` and `TracerProvider` here after the code sample. I think that this can also be expounded upon in the manual tracing article separately, since that is intended to go into a little more depth about these topics. Let me know what you think!

Reply:
Looks good